### PR TITLE
Pass explicit -qt5 to all moc and qmake calls.

### PR DESCRIPTION
**Pull request by jmallach**
Some systems like Debian, Ubuntu and other derivatives, expect qtchooser
will be used to select the appropriate Qt version, as Qt5 and Qt4 can
co-exist.

As MAME's build system does calls to moc and qmake directly, and Qt5 is
now the required version for the debugger, pass -qt5 to all moc and qmake
calls to avoid the build fail.

### CODE REVIEW

**Comment by jmallach** [comment]
This patch has been tested on Debian builds.

**Comment by mmicko** [comment]
Thanks, Think we need to confirm osx builds with QT5 works as well, but
anyway all params should be same on all platforms

**Comment by mmicko** [comment]
Reverted

**Comment by mmicko** [comment]
This change breaks compiling on Fedora/RHEL/CentOS.

**Comment by jmallach** [comment]
Hmm. I'll see if I can work with the Debian Qt maintainers for a version that does not break others. I had understood this was it.

**Comment by galibert** [comment]
The problem is that non-qtchooser moc does not accept -qt5, you get the
(logical) error "Unknown options: q, t, 5.".  Debian not making moc-qt5
visible in the path is batshit stupid in the first place.

  OG.

**Comment by jmallach** [comment]
@mmicko after chatting to the Qt maintainer in Debian/Ubuntu, I've learned the following:

 - apparently Qt upstream explicitly asked Debian to not ship moc-qt4 or moc-qt5 in the path, but use qtchooser instead. Debian had been shipping it like Fedora until they decided to follow what upstream was asking.

 - as there are two diverging views on how this needs to be installed and what will actually be available on Debian/RH systems, I guess we need to support both. Can we do a test for qtchooser, and do one thing or the other depending on the result?

I suggest:

 - test for the return value of ``readlink -f `which qmake` | grep -i -q qtchooser``.
 - If it returns 0, populate a variable with "-qt5" and add the variable to all qmake and moc calls like in the patch. If the variable is empty, no behaviour changes, and if the variable isn't empty, use the value as an argument to moc and qmake.

I can try to cut a patch myself, but probably my Lua is silly enough that it'd be a magnitude easier for you to do it. :P

**Comment by jmallach** [comment]
@mmicko, ugh. It seems it's worse than this.

In https://github.com/mamedev/mame/blob/master/scripts/src/osd/modules.lua#L296, there's a full test for "QT_HOME". What is this? Qt knows about this variable, or at least ``git grep QT_HOME`` in qtbase source code returns nothing.

Also, in order to find moc, it's path should be queried to qmake instead, with something like ``qmake -qt5 -query QT_INSTALL_BINS`` (I am unsure of the Fedora version of this command).

And there are some hardcoded paths in here, like ``/bin/qmake``.

In short, there's enough suspicious bits in here that I don't know how to proceed. I formally request help here. :)
